lib/cc/workspace.rb

```diff
@@ -0,0 +1,66 @@
+module CC
+  class Workspace
+    autoload :Exclusions, "cc/workspace/exclusions"
+    autoload :PathsValidator, "cc/workspace/paths_validator"
+
+    def initialize(paths: nil, prefix: "")
+      @prefix = prefix
+
+      if paths.present?
+        validator = PathsValidator.new(paths)
+        validator.run
+
+        @paths = paths
+      end
+
+      CLI.debug("workspace initialize", prefix: prefix)
+    end
+
+    def paths
+      @paths || ["./"]
+    end
+
+    def filter(exclude_paths)
+      return self unless exclude_paths.present?
+
+      exclusions = Exclusions.new(exclude_paths)
+
+      CLI.debug("workspace filter start", exclude_paths: exclude_paths)
+
+      @paths ||= Dir.glob("*", File::FNM_DOTMATCH)
+      @paths = @paths.flat_map { |name| expand(name, exclusions) }
+
+      CLI.debug("workspace filter end", paths: paths)
+
+      self
+    end
+
+    private
+
+    attr_reader :prefix
+
+    def expand(name, exclusions)
+      path = "#{prefix}#{name}"
+
+      return [] if %w[. .. .git].include?(name)
```

Comment:
This series of short-circuited returns is hard to read to me. WDYT of using `if...else`?